### PR TITLE
Fixed CPT-58: Meta tags issue

### PR DESCRIPTION
**Pull request by maxim-babenkocommerce**

### CODE REVIEW

File: Model/TieManagement.php [hunk 1]
```diff
@@ -183,25 +202,24 @@ public function updateCmsLinks($currentPageId, $linksArray)
...
-            $linkedPageId = $this->getLinkedCmsIdByStoreId($currentPageId, $targetStoreId);
-            $page = $this->pageRepository->getById($linkedPageId);
+            $page = $this->pageRepository->getById($pageId);
             $result = $page->getIdentifier();
         } catch (\Exception $e) {
-            $result = "";
+            $result = '';
         }
-        return $result;
+        return !empty($result) ? $result : '';
```
Comment (vsevolod-tarasov) [inline Model/TieManagement.php]: there is no need fir empty check here, result will be populated in any case

File: Plugin/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php [hunk 2]
```diff
@@ -108,12 +108,11 @@ public function afterSwitch(StoreRewriteUrl $subject, $result, $targetStore, $ol
             } while (!isset($currentPageIdentifier));
 
             if (isset($currentPageIdentifier)) {
-                $linkedPageName = $this->tieManagement->getLinkedCmsKey($currentPageIdentifier, $targetStoreId);
                 $linkedPageId = $this->tieManagement->getLinkedCmsIdByStoreId($currentPageIdentifier, $targetStoreId);
-
                 if (isset($linkedPageId) && $linkedPageId != 0) {
                     $this->urlBuilder->setScope($targetStoreId);
-                    $result = $this->urlBuilder->getUrl(null, ['_direct' => $linkedPageName]);
+                    $result = $this->urlBuilder
```
Comment (vsevolod-tarasov) [inline Plugin/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php]: Is this functionality related to meta tags?

File: Model/TieManagement.php [hunk 3]
```diff
@@ -103,37 +103,56 @@ public function __construct(
...
-                $this->urlInterface->setScope($targetStoreId);
+        foreach ($currentPageStores as $currentPageStore) {
+            $linkedPages[] = [
+                'page_id' => $currentPageId,
+                'linked_page_id' => $currentPageId,
+                'store_id' => $currentPageStore,
+            ];
+        }
+        foreach ($linkedPages as $linkedPage) {
+            $targetStoreId = (int)$linkedPage['store_id'];
```
Comment (vsevolod-tarasov) [inline Model/TieManagement.php]: the space is required after (int)

File: Model/TieManagement.php [hunk 4]
```diff
@@ -155,7 +174,7 @@ public function updateCmsLinks($currentPageId, $linksArray)
     {
         $currentPageInfo = [
             'stores' => $this->cmsPageModel->lookupStoreIds($currentPageId),
-            'page_id' => (int) $currentPageId
+            'page_id' => (int) $currentPageId,
```
Comment (vsevolod-tarasov) [inline Model/TieManagement.php]: is that functionality related to Meta tags issue?

File: Model/TieManagement.php (hunk 1 above)
Comment (maxim-babenkocommerce) [reply Model/TieManagement.php]: it waits for a string in return, but null can be in $result

File: Plugin/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php (hunk 2 above)
Comment (maxim-babenkocommerce) [reply Plugin/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php]: I changed the method for the current task, so I changed the places where this method is called

File: Model/TieManagement.php (hunk 4 above)
Comment (maxim-babenkocommerce) [reply Model/TieManagement.php]: just refactor nearby places for uniformity

File: Model/TieManagement.php (hunk 1 above)
Comment (vsevolod-tarasov) [reply Model/TieManagement.php]: Anyway there is no need for empty check
